{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd \n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt \n",
    "import seaborn as sn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import geoai\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: geoai-py in ./.venv/lib/python3.13/site-packages (0.4.3)\n",
      "Requirement already satisfied: albumentations in ./.venv/lib/python3.13/site-packages (from geoai-py) (2.0.5)\n",
      "Requirement already satisfied: contextily in ./.venv/lib/python3.13/site-packages (from geoai-py) (1.6.2)\n",
      "Requirement already satisfied: geopandas in ./.venv/lib/python3.13/site-packages (from geoai-py) (1.0.1)\n",
      "Requirement already satisfied: huggingface_hub in ./.venv/lib/python3.13/site-packages (from geoai-py) (0.29.3)\n",
      "Requirement already satisfied: jupyter-server-proxy in ./.venv/lib/python3.13/site-packages (from geoai-py) (4.4.0)\n",
      "Requirement already satisfied: leafmap in ./.venv/lib/python3.13/site-packages (from geoai-py) (0.42.13)\n",
      "Requirement already satisfied: localtileserver in ./.venv/lib/python3.13/site-packages (from geoai-py) (0.10.6)\n",
      "Requirement already satisfied: mapclassify in ./.venv/lib/python3.13/site-packages (from geoai-py) (2.8.1)\n",
      "Requirement already satisfied: overturemaps in ./.venv/lib/python3.13/site-packages (from geoai-py) (0.14.0)\n",
      "Requirement already satisfied: planetary-computer in ./.venv/lib/python3.13/site-packages (from geoai-py) (1.0.0)\n",
      "Requirement already satisfied: pystac-client in ./.venv/lib/python3.13/site-packages (from geoai-py) (0.8.6)\n",
      "Requirement already satisfied: rasterio in ./.venv/lib/python3.13/site-packages (from geoai-py) (1.4.3)\n",
      "Requirement already satisfied: rioxarray in ./.venv/lib/python3.13/site-packages (from geoai-py) (0.18.2)\n",
      "Requirement already satisfied: scikit-image in ./.venv/lib/python3.13/site-packages (from geoai-py) (0.25.2)\n",
      "Requirement already satisfied: scikit-learn in ./.venv/lib/python3.13/site-packages (from geoai-py) (1.6.1)\n",
      "Requirement already satisfied: torch in ./.venv/lib/python3.13/site-packages (from geoai-py) (2.6.0)\n",
      "Requirement already satisfied: torchgeo in ./.venv/lib/python3.13/site-packages (from geoai-py) (0.6.2)\n",
      "Requirement already satisfied: torchinfo in ./.venv/lib/python3.13/site-packages (from geoai-py) (1.8.0)\n",
      "Requirement already satisfied: tqdm in ./.venv/lib/python3.13/site-packages (from geoai-py) (4.67.1)\n",
      "Requirement already satisfied: transformers in ./.venv/lib/python3.13/site-packages (from geoai-py) (4.50.0)\n",
      "Requirement already satisfied: numpy>=1.24.4 in ./.venv/lib/python3.13/site-packages (from albumentations->geoai-py) (2.2.4)\n",
      "Requirement already satisfied: scipy>=1.10.0 in ./.venv/lib/python3.13/site-packages (from albumentations->geoai-py) (1.15.2)\n",
      "Requirement already satisfied: PyYAML in ./.venv/lib/python3.13/site-packages (from albumentations->geoai-py) (6.0.2)\n",
      "Requirement already satisfied: pydantic>=2.9.2 in ./.venv/lib/python3.13/site-packages (from albumentations->geoai-py) (2.10.6)\n",
      "Requirement already satisfied: albucore==0.0.23 in ./.venv/lib/python3.13/site-packages (from albumentations->geoai-py) (0.0.23)\n",
      "Requirement already satisfied: opencv-python-headless>=4.9.0.80 in ./.venv/lib/python3.13/site-packages (from albumentations->geoai-py) (4.11.0.86)\n",
      "Requirement already satisfied: stringzilla>=3.10.4 in ./.venv/lib/python3.13/site-packages (from albucore==0.0.23->albumentations->geoai-py) (3.12.3)\n",
      "Requirement already satisfied: simsimd>=5.9.2 in ./.venv/lib/python3.13/site-packages (from albucore==0.0.23->albumentations->geoai-py) (6.2.1)\n",
      "Requirement already satisfied: geopy in ./.venv/lib/python3.13/site-packages (from contextily->geoai-py) (2.4.1)\n",
      "Requirement already satisfied: matplotlib in ./.venv/lib/python3.13/site-packages (from contextily->geoai-py) (3.10.1)\n",
      "Requirement already satisfied: mercantile in ./.venv/lib/python3.13/site-packages (from contextily->geoai-py) (1.2.1)\n",
      "Requirement already satisfied: pillow in ./.venv/lib/python3.13/site-packages (from contextily->geoai-py) (11.1.0)\n",
      "Requirement already satisfied: requests in ./.venv/lib/python3.13/site-packages (from contextily->geoai-py) (2.32.3)\n",
      "Requirement already satisfied: joblib in ./.venv/lib/python3.13/site-packages (from contextily->geoai-py) (1.4.2)\n",
      "Requirement already satisfied: xyzservices in ./.venv/lib/python3.13/site-packages (from contextily->geoai-py) (2025.1.0)\n",
      "Requirement already satisfied: pyogrio>=0.7.2 in ./.venv/lib/python3.13/site-packages (from geopandas->geoai-py) (0.10.0)\n",
      "Requirement already satisfied: packaging in ./.venv/lib/python3.13/site-packages (from geopandas->geoai-py) (24.2)\n",
      "Requirement already satisfied: pandas>=1.4.0 in ./.venv/lib/python3.13/site-packages (from geopandas->geoai-py) (2.2.3)\n",
      "Requirement already satisfied: pyproj>=3.3.0 in ./.venv/lib/python3.13/site-packages (from geopandas->geoai-py) (3.7.1)\n",
      "Requirement already satisfied: shapely>=2.0.0 in ./.venv/lib/python3.13/site-packages (from geopandas->geoai-py) (2.0.7)\n",
      "Requirement already satisfied: filelock in ./.venv/lib/python3.13/site-packages (from huggingface_hub->geoai-py) (3.18.0)\n",
      "Requirement already satisfied: fsspec>=2023.5.0 in ./.venv/lib/python3.13/site-packages (from huggingface_hub->geoai-py) (2025.3.0)\n",
      "Requirement already satisfied: typing-extensions>=3.7.4.3 in ./.venv/lib/python3.13/site-packages (from huggingface_hub->geoai-py) (4.12.2)\n",
      "Requirement already satisfied: aiohttp in ./.venv/lib/python3.13/site-packages (from jupyter-server-proxy->geoai-py) (3.11.14)\n",
      "Requirement already satisfied: jupyter-server>=1.24.0 in ./.venv/lib/python3.13/site-packages (from jupyter-server-proxy->geoai-py) (2.15.0)\n",
      "Requirement already satisfied: simpervisor>=1.0.0 in ./.venv/lib/python3.13/site-packages (from jupyter-server-proxy->geoai-py) (1.0.0)\n",
      "Requirement already satisfied: tornado>=6.1.0 in ./.venv/lib/python3.13/site-packages (from jupyter-server-proxy->geoai-py) (6.4.2)\n",
      "Requirement already satisfied: traitlets>=5.1.0 in ./.venv/lib/python3.13/site-packages (from jupyter-server-proxy->geoai-py) (5.14.3)\n",
      "Requirement already satisfied: anywidget in ./.venv/lib/python3.13/site-packages (from leafmap->geoai-py) (0.9.17)\n",
      "Requirement already satisfied: bqplot in ./.venv/lib/python3.13/site-packages (from leafmap->geoai-py) (0.12.44)\n",
      "Requirement already satisfied: colour in ./.venv/lib/python3.13/site-packages (from leafmap->geoai-py) (0.1.5)\n",
      "Requirement already satisfied: folium in ./.venv/lib/python3.13/site-packages (from leafmap->geoai-py) (0.19.5)\n",
      "Requirement already satisfied: gdown in ./.venv/lib/python3.13/site-packages (from leafmap->geoai-py) (5.2.0)\n",
      "Requirement already satisfied: geojson in ./.venv/lib/python3.13/site-packages (from leafmap->geoai-py) (3.2.0)\n",
      "Requirement already satisfied: ipyevents in ./.venv/lib/python3.13/site-packages (from leafmap->geoai-py) (2.0.2)\n",
      "Requirement already satisfied: ipyfilechooser in ./.venv/lib/python3.13/site-packages (from leafmap->geoai-py) (0.6.0)\n",
      "Requirement already satisfied: ipyleaflet in ./.venv/lib/python3.13/site-packages (from leafmap->geoai-py) (0.19.2)\n",
      "Requirement already satisfied: ipyvuetify in ./.venv/lib/python3.13/site-packages (from leafmap->geoai-py) (1.11.1)\n",
      "Requirement already satisfied: ipywidgets in ./.venv/lib/python3.13/site-packages (from leafmap->geoai-py) (8.1.5)\n",
      "Requirement already satisfied: plotly in ./.venv/lib/python3.13/site-packages (from leafmap->geoai-py) (6.0.1)\n",
      "Requirement already satisfied: psutil in ./.venv/lib/python3.13/site-packages (from leafmap->geoai-py) (7.0.0)\n",
      "Requirement already satisfied: pyshp in ./.venv/lib/python3.13/site-packages (from leafmap->geoai-py) (2.3.1)\n",
      "Requirement already satisfied: python-box in ./.venv/lib/python3.13/site-packages (from leafmap->geoai-py) (7.3.2)\n",
      "Requirement already satisfied: scooby in ./.venv/lib/python3.13/site-packages (from leafmap->geoai-py) (0.10.0)\n",
      "Requirement already satisfied: whiteboxgui in ./.venv/lib/python3.13/site-packages (from leafmap->geoai-py) (2.3.0)\n",
      "Requirement already satisfied: click in ./.venv/lib/python3.13/site-packages (from localtileserver->geoai-py) (8.1.8)\n",
      "Requirement already satisfied: flask<4,>=2.0.0 in ./.venv/lib/python3.13/site-packages (from localtileserver->geoai-py) (3.1.0)\n",
      "Requirement already satisfied: Flask-Caching in ./.venv/lib/python3.13/site-packages (from localtileserver->geoai-py) (2.3.1)\n",
      "Requirement already satisfied: flask-cors in ./.venv/lib/python3.13/site-packages (from localtileserver->geoai-py) (5.0.1)\n",
      "Requirement already satisfied: flask-restx>=1.3.0 in ./.venv/lib/python3.13/site-packages (from localtileserver->geoai-py) (1.3.0)\n",
      "Requirement already satisfied: rio-tiler in ./.venv/lib/python3.13/site-packages (from localtileserver->geoai-py) (7.5.1)\n",
      "Requirement already satisfied: rio-cogeo in ./.venv/lib/python3.13/site-packages (from localtileserver->geoai-py) (5.4.1)\n",
      "Requirement already satisfied: server-thread in ./.venv/lib/python3.13/site-packages (from localtileserver->geoai-py) (0.3.0)\n",
      "Requirement already satisfied: werkzeug in ./.venv/lib/python3.13/site-packages (from localtileserver->geoai-py) (3.1.3)\n",
      "Requirement already satisfied: networkx>=2.7 in ./.venv/lib/python3.13/site-packages (from mapclassify->geoai-py) (3.4.2)\n",
      "Requirement already satisfied: threadpoolctl>=3.1.0 in ./.venv/lib/python3.13/site-packages (from scikit-learn->geoai-py) (3.6.0)\n",
      "Requirement already satisfied: pyarrow>=15.0.2 in ./.venv/lib/python3.13/site-packages (from overturemaps->geoai-py) (19.0.1)\n",
      "Requirement already satisfied: pystac>=1.0.0 in ./.venv/lib/python3.13/site-packages (from planetary-computer->geoai-py) (1.12.2)\n",
      "Requirement already satisfied: pytz>=2020.5 in ./.venv/lib/python3.13/site-packages (from planetary-computer->geoai-py) (2025.1)\n",
      "Requirement already satisfied: python-dotenv in ./.venv/lib/python3.13/site-packages (from planetary-computer->geoai-py) (1.0.1)\n",
      "Requirement already satisfied: python-dateutil>=2.8.2 in ./.venv/lib/python3.13/site-packages (from pystac-client->geoai-py) (2.9.0.post0)\n",
      "Requirement already satisfied: affine in ./.venv/lib/python3.13/site-packages (from rasterio->geoai-py) (2.4.0)\n",
      "Requirement already satisfied: attrs in ./.venv/lib/python3.13/site-packages (from rasterio->geoai-py) (25.3.0)\n",
      "Requirement already satisfied: certifi in ./.venv/lib/python3.13/site-packages (from rasterio->geoai-py) (2025.1.31)\n",
      "Requirement already satisfied: cligj>=0.5 in ./.venv/lib/python3.13/site-packages (from rasterio->geoai-py) (0.7.2)\n",
      "Requirement already satisfied: click-plugins in ./.venv/lib/python3.13/site-packages (from rasterio->geoai-py) (1.1.1)\n",
      "Requirement already satisfied: pyparsing in ./.venv/lib/python3.13/site-packages (from rasterio->geoai-py) (3.2.1)\n",
      "Requirement already satisfied: xarray>=2024.7.0 in ./.venv/lib/python3.13/site-packages (from rioxarray->geoai-py) (2025.3.0)\n",
      "Requirement already satisfied: imageio!=2.35.0,>=2.33 in ./.venv/lib/python3.13/site-packages (from scikit-image->geoai-py) (2.37.0)\n",
      "Requirement already satisfied: tifffile>=2022.8.12 in ./.venv/lib/python3.13/site-packages (from scikit-image->geoai-py) (2025.3.13)\n",
      "Requirement already satisfied: lazy-loader>=0.4 in ./.venv/lib/python3.13/site-packages (from scikit-image->geoai-py) (0.4)\n",
      "Requirement already satisfied: jinja2 in ./.venv/lib/python3.13/site-packages (from torch->geoai-py) (3.1.6)\n",
      "Requirement already satisfied: setuptools in ./.venv/lib/python3.13/site-packages (from torch->geoai-py) (76.0.0)\n",
      "Requirement already satisfied: sympy==1.13.1 in ./.venv/lib/python3.13/site-packages (from torch->geoai-py) (1.13.1)\n",
      "Requirement already satisfied: mpmath<1.4,>=1.1.0 in ./.venv/lib/python3.13/site-packages (from sympy==1.13.1->torch->geoai-py) (1.3.0)\n",
      "Requirement already satisfied: einops>=0.3 in ./.venv/lib/python3.13/site-packages (from torchgeo->geoai-py) (0.8.1)\n",
      "Requirement already satisfied: fiona>=1.8.21 in ./.venv/lib/python3.13/site-packages (from torchgeo->geoai-py) (1.10.1)\n",
      "Requirement already satisfied: kornia>=0.7.3 in ./.venv/lib/python3.13/site-packages (from torchgeo->geoai-py) (0.8.0)\n",
      "Requirement already satisfied: lightly!=1.4.26,>=1.4.5 in ./.venv/lib/python3.13/site-packages (from torchgeo->geoai-py) (1.5.19)\n",
      "Requirement already satisfied: lightning!=2.3.*,>=2 in ./.venv/lib/python3.13/site-packages (from lightning[pytorch-extra]!=2.3.*,>=2->torchgeo->geoai-py) (2.5.1)\n",
      "Requirement already satisfied: rtree>=1 in ./.venv/lib/python3.13/site-packages (from torchgeo->geoai-py) (1.4.0)\n",
      "Requirement already satisfied: segmentation-models-pytorch>=0.2 in ./.venv/lib/python3.13/site-packages (from torchgeo->geoai-py) (0.4.0)\n",
      "Requirement already satisfied: timm>=0.4.12 in ./.venv/lib/python3.13/site-packages (from torchgeo->geoai-py) (1.0.15)\n",
      "Requirement already satisfied: torchmetrics>=0.10 in ./.venv/lib/python3.13/site-packages (from torchgeo->geoai-py) (1.7.0)\n",
      "Requirement already satisfied: torchvision>=0.14 in ./.venv/lib/python3.13/site-packages (from torchgeo->geoai-py) (0.21.0)\n",
      "Requirement already satisfied: regex!=2019.12.17 in ./.venv/lib/python3.13/site-packages (from transformers->geoai-py) (2024.11.6)\n",
      "Requirement already satisfied: tokenizers<0.22,>=0.21 in ./.venv/lib/python3.13/site-packages (from transformers->geoai-py) (0.21.1)\n",
      "Requirement already satisfied: safetensors>=0.4.3 in ./.venv/lib/python3.13/site-packages (from transformers->geoai-py) (0.5.3)\n",
      "Requirement already satisfied: itsdangerous>=2.2 in ./.venv/lib/python3.13/site-packages (from flask<4,>=2.0.0->localtileserver->geoai-py) (2.2.0)\n",
      "Requirement already satisfied: blinker>=1.9 in ./.venv/lib/python3.13/site-packages (from flask<4,>=2.0.0->localtileserver->geoai-py) (1.9.0)\n",
      "Requirement already satisfied: aniso8601>=0.82 in ./.venv/lib/python3.13/site-packages (from flask-restx>=1.3.0->localtileserver->geoai-py) (10.0.0)\n",
      "Requirement already satisfied: jsonschema in ./.venv/lib/python3.13/site-packages (from flask-restx>=1.3.0->localtileserver->geoai-py) (4.23.0)\n",
      "Requirement already satisfied: importlib-resources in ./.venv/lib/python3.13/site-packages (from flask-restx>=1.3.0->localtileserver->geoai-py) (6.5.2)\n",
      "Requirement already satisfied: MarkupSafe>=2.0 in ./.venv/lib/python3.13/site-packages (from jinja2->torch->geoai-py) (3.0.2)\n",
      "Requirement already satisfied: anyio>=3.1.0 in ./.venv/lib/python3.13/site-packages (from jupyter-server>=1.24.0->jupyter-server-proxy->geoai-py) (4.9.0)\n",
      "Requirement already satisfied: argon2-cffi>=21.1 in ./.venv/lib/python3.13/site-packages (from jupyter-server>=1.24.0->jupyter-server-proxy->geoai-py) (23.1.0)\n",
      "Requirement already satisfied: jupyter-client>=7.4.4 in ./.venv/lib/python3.13/site-packages (from jupyter-server>=1.24.0->jupyter-server-proxy->geoai-py) (8.6.3)\n",
      "Requirement already satisfied: jupyter-core!=5.0.*,>=4.12 in ./.venv/lib/python3.13/site-packages (from jupyter-server>=1.24.0->jupyter-server-proxy->geoai-py) (5.7.2)\n",
      "Requirement already satisfied: jupyter-events>=0.11.0 in ./.venv/lib/python3.13/site-packages (from jupyter-server>=1.24.0->jupyter-server-proxy->geoai-py) (0.12.0)\n",
      "Requirement already satisfied: jupyter-server-terminals>=0.4.4 in ./.venv/lib/python3.13/site-packages (from jupyter-server>=1.24.0->jupyter-server-proxy->geoai-py) (0.5.3)\n",
      "Requirement already satisfied: nbconvert>=6.4.4 in ./.venv/lib/python3.13/site-packages (from jupyter-server>=1.24.0->jupyter-server-proxy->geoai-py) (7.16.6)\n",
      "Requirement already satisfied: nbformat>=5.3.0 in ./.venv/lib/python3.13/site-packages (from jupyter-server>=1.24.0->jupyter-server-proxy->geoai-py) (5.10.4)\n",
      "Requirement already satisfied: overrides>=5.0 in ./.venv/lib/python3.13/site-packages (from jupyter-server>=1.24.0->jupyter-server-proxy->geoai-py) (7.7.0)\n",
      "Requirement already satisfied: prometheus-client>=0.9 in ./.venv/lib/python3.13/site-packages (from jupyter-server>=1.24.0->jupyter-server-proxy->geoai-py) (0.21.1)\n",
      "Requirement already satisfied: pyzmq>=24 in ./.venv/lib/python3.13/site-packages (from jupyter-server>=1.24.0->jupyter-server-proxy->geoai-py) (26.3.0)\n",
      "Requirement already satisfied: send2trash>=1.8.2 in ./.venv/lib/python3.13/site-packages (from jupyter-server>=1.24.0->jupyter-server-proxy->geoai-py) (1.8.3)\n",
      "Requirement already satisfied: terminado>=0.8.3 in ./.venv/lib/python3.13/site-packages (from jupyter-server>=1.24.0->jupyter-server-proxy->geoai-py) (0.18.1)\n",
      "Requirement already satisfied: websocket-client>=1.7 in ./.venv/lib/python3.13/site-packages (from jupyter-server>=1.24.0->jupyter-server-proxy->geoai-py) (1.8.0)\n",
      "Requirement already satisfied: kornia_rs>=0.1.0 in ./.venv/lib/python3.13/site-packages (from kornia>=0.7.3->torchgeo->geoai-py) (0.1.8)\n",
      "Requirement already satisfied: hydra-core>=1.0.0 in ./.venv/lib/python3.13/site-packages (from lightly!=1.4.26,>=1.4.5->torchgeo->geoai-py) (1.3.2)\n",
      "Requirement already satisfied: lightly_utils~=0.0.0 in ./.venv/lib/python3.13/site-packages (from lightly!=1.4.26,>=1.4.5->torchgeo->geoai-py) (0.0.2)\n",
      "Requirement already satisfied: six>=1.10 in ./.venv/lib/python3.13/site-packages (from lightly!=1.4.26,>=1.4.5->torchgeo->geoai-py) (1.17.0)\n",
      "Requirement already satisfied: pytorch_lightning>=1.0.4 in ./.venv/lib/python3.13/site-packages (from lightly!=1.4.26,>=1.4.5->torchgeo->geoai-py) (2.5.1)\n",
      "Requirement already satisfied: urllib3>=1.25.3 in ./.venv/lib/python3.13/site-packages (from lightly!=1.4.26,>=1.4.5->torchgeo->geoai-py) (2.3.0)\n",
      "Requirement already satisfied: aenum>=3.1.11 in ./.venv/lib/python3.13/site-packages (from lightly!=1.4.26,>=1.4.5->torchgeo->geoai-py) (3.1.15)\n",
      "Requirement already satisfied: lightning-utilities<2.0,>=0.10.0 in ./.venv/lib/python3.13/site-packages (from lightning!=2.3.*,>=2->lightning[pytorch-extra]!=2.3.*,>=2->torchgeo->geoai-py) (0.14.2)\n",
      "Requirement already satisfied: jsonargparse<5.0,>=4.27.7 in ./.venv/lib/python3.13/site-packages (from jsonargparse[signatures]<5.0,>=4.27.7; extra == \"pytorch-extra\"->lightning[pytorch-extra]!=2.3.*,>=2->torchgeo->geoai-py) (4.37.0)\n",
      "Requirement already satisfied: omegaconf<3.0,>=2.2.3 in ./.venv/lib/python3.13/site-packages (from lightning[pytorch-extra]!=2.3.*,>=2->torchgeo->geoai-py) (2.3.0)\n",
      "Requirement already satisfied: rich<14.0,>=12.3.0 in ./.venv/lib/python3.13/site-packages (from lightning[pytorch-extra]!=2.3.*,>=2->torchgeo->geoai-py) (13.9.4)\n",
      "Requirement already satisfied: tensorboardX<3.0,>=2.2 in ./.venv/lib/python3.13/site-packages (from lightning[pytorch-extra]!=2.3.*,>=2->torchgeo->geoai-py) (2.6.2.2)\n",
      "Requirement already satisfied: contourpy>=1.0.1 in ./.venv/lib/python3.13/site-packages (from matplotlib->contextily->geoai-py) (1.3.1)\n",
      "Requirement already satisfied: cycler>=0.10 in ./.venv/lib/python3.13/site-packages (from matplotlib->contextily->geoai-py) (0.12.1)\n",
      "Requirement already satisfied: fonttools>=4.22.0 in ./.venv/lib/python3.13/site-packages (from matplotlib->contextily->geoai-py) (4.56.0)\n",
      "Requirement already satisfied: kiwisolver>=1.3.1 in ./.venv/lib/python3.13/site-packages (from matplotlib->contextily->geoai-py) (1.4.8)\n",
      "Requirement already satisfied: tzdata>=2022.7 in ./.venv/lib/python3.13/site-packages (from pandas>=1.4.0->geopandas->geoai-py) (2025.1)\n",
      "Requirement already satisfied: annotated-types>=0.6.0 in ./.venv/lib/python3.13/site-packages (from pydantic>=2.9.2->albumentations->geoai-py) (0.7.0)\n",
      "Requirement already satisfied: pydantic-core==2.27.2 in ./.venv/lib/python3.13/site-packages (from pydantic>=2.9.2->albumentations->geoai-py) (2.27.2)\n",
      "Requirement already satisfied: charset-normalizer<4,>=2 in ./.venv/lib/python3.13/site-packages (from requests->contextily->geoai-py) (3.4.1)\n",
      "Requirement already satisfied: idna<4,>=2.5 in ./.venv/lib/python3.13/site-packages (from requests->contextily->geoai-py) (3.10)\n",
      "Requirement already satisfied: efficientnet-pytorch>=0.6.1 in ./.venv/lib/python3.13/site-packages (from segmentation-models-pytorch>=0.2->torchgeo->geoai-py) (0.7.1)\n",
      "Requirement already satisfied: pretrainedmodels>=0.7.1 in ./.venv/lib/python3.13/site-packages (from segmentation-models-pytorch>=0.2->torchgeo->geoai-py) (0.7.4)\n",
      "Requirement already satisfied: aiohappyeyeballs>=2.3.0 in ./.venv/lib/python3.13/site-packages (from aiohttp->jupyter-server-proxy->geoai-py) (2.6.1)\n",
      "Requirement already satisfied: aiosignal>=1.1.2 in ./.venv/lib/python3.13/site-packages (from aiohttp->jupyter-server-proxy->geoai-py) (1.3.2)\n",
      "Requirement already satisfied: frozenlist>=1.1.1 in ./.venv/lib/python3.13/site-packages (from aiohttp->jupyter-server-proxy->geoai-py) (1.5.0)\n",
      "Requirement already satisfied: multidict<7.0,>=4.5 in ./.venv/lib/python3.13/site-packages (from aiohttp->jupyter-server-proxy->geoai-py) (6.2.0)\n",
      "Requirement already satisfied: propcache>=0.2.0 in ./.venv/lib/python3.13/site-packages (from aiohttp->jupyter-server-proxy->geoai-py) (0.3.0)\n",
      "Requirement already satisfied: yarl<2.0,>=1.17.0 in ./.venv/lib/python3.13/site-packages (from aiohttp->jupyter-server-proxy->geoai-py) (1.18.3)\n",
      "Requirement already satisfied: psygnal>=0.8.1 in ./.venv/lib/python3.13/site-packages (from anywidget->leafmap->geoai-py) (0.12.0)\n",
      "Requirement already satisfied: comm>=0.1.3 in ./.venv/lib/python3.13/site-packages (from ipywidgets->leafmap->geoai-py) (0.2.2)\n",
      "Requirement already satisfied: ipython>=6.1.0 in ./.venv/lib/python3.13/site-packages (from ipywidgets->leafmap->geoai-py) (9.0.2)\n",
      "Requirement already satisfied: widgetsnbextension~=4.0.12 in ./.venv/lib/python3.13/site-packages (from ipywidgets->leafmap->geoai-py) (4.0.13)\n",
      "Requirement already satisfied: jupyterlab-widgets~=3.0.12 in ./.venv/lib/python3.13/site-packages (from ipywidgets->leafmap->geoai-py) (3.0.13)\n",
      "Requirement already satisfied: traittypes>=0.0.6 in ./.venv/lib/python3.13/site-packages (from bqplot->leafmap->geoai-py) (0.2.1)\n",
      "Requirement already satisfied: cachelib>=0.9.0 in ./.venv/lib/python3.13/site-packages (from Flask-Caching->localtileserver->geoai-py) (0.13.0)\n",
      "Requirement already satisfied: branca>=0.6.0 in ./.venv/lib/python3.13/site-packages (from folium->leafmap->geoai-py) (0.8.1)\n",
      "Requirement already satisfied: beautifulsoup4 in ./.venv/lib/python3.13/site-packages (from gdown->leafmap->geoai-py) (4.13.3)\n",
      "Requirement already satisfied: geographiclib<3,>=1.52 in ./.venv/lib/python3.13/site-packages (from geopy->contextily->geoai-py) (2.0)\n",
      "Requirement already satisfied: jupyter-leaflet<0.20,>=0.19 in ./.venv/lib/python3.13/site-packages (from ipyleaflet->leafmap->geoai-py) (0.19.2)\n",
      "Requirement already satisfied: ipyvue<2,>=1.7 in ./.venv/lib/python3.13/site-packages (from ipyvuetify->leafmap->geoai-py) (1.11.2)\n",
      "Requirement already satisfied: narwhals>=1.15.1 in ./.venv/lib/python3.13/site-packages (from plotly->leafmap->geoai-py) (1.31.0)\n",
      "Requirement already satisfied: morecantile<7.0,>=5.0 in ./.venv/lib/python3.13/site-packages (from rio-cogeo->localtileserver->geoai-py) (6.2.0)\n",
      "Requirement already satisfied: cachetools in ./.venv/lib/python3.13/site-packages (from rio-tiler->localtileserver->geoai-py) (5.5.2)\n",
      "Requirement already satisfied: color-operations in ./.venv/lib/python3.13/site-packages (from rio-tiler->localtileserver->geoai-py) (0.1.6)\n",
      "Requirement already satisfied: httpx in ./.venv/lib/python3.13/site-packages (from rio-tiler->localtileserver->geoai-py) (0.28.1)\n",
      "Requirement already satisfied: numexpr in ./.venv/lib/python3.13/site-packages (from rio-tiler->localtileserver->geoai-py) (2.10.2)\n",
      "Requirement already satisfied: uvicorn in ./.venv/lib/python3.13/site-packages (from server-thread->localtileserver->geoai-py) (0.34.0)\n",
      "Requirement already satisfied: ipytree in ./.venv/lib/python3.13/site-packages (from whiteboxgui->leafmap->geoai-py) (0.2.2)\n",
      "Requirement already satisfied: whitebox in ./.venv/lib/python3.13/site-packages (from whiteboxgui->leafmap->geoai-py) (2.3.6)\n",
      "Requirement already satisfied: sniffio>=1.1 in ./.venv/lib/python3.13/site-packages (from anyio>=3.1.0->jupyter-server>=1.24.0->jupyter-server-proxy->geoai-py) (1.3.1)\n",
      "Requirement already satisfied: argon2-cffi-bindings in ./.venv/lib/python3.13/site-packages (from argon2-cffi>=21.1->jupyter-server>=1.24.0->jupyter-server-proxy->geoai-py) (21.2.0)\n",
      "Requirement already satisfied: antlr4-python3-runtime==4.9.* in ./.venv/lib/python3.13/site-packages (from hydra-core>=1.0.0->lightly!=1.4.26,>=1.4.5->torchgeo->geoai-py) (4.9.3)\n",
      "Requirement already satisfied: decorator in ./.venv/lib/python3.13/site-packages (from ipython>=6.1.0->ipywidgets->leafmap->geoai-py) (5.2.1)\n",
      "Requirement already satisfied: ipython-pygments-lexers in ./.venv/lib/python3.13/site-packages (from ipython>=6.1.0->ipywidgets->leafmap->geoai-py) (1.1.1)\n",
      "Requirement already satisfied: jedi>=0.16 in ./.venv/lib/python3.13/site-packages (from ipython>=6.1.0->ipywidgets->leafmap->geoai-py) (0.19.2)\n",
      "Requirement already satisfied: matplotlib-inline in ./.venv/lib/python3.13/site-packages (from ipython>=6.1.0->ipywidgets->leafmap->geoai-py) (0.1.7)\n",
      "Requirement already satisfied: pexpect>4.3 in ./.venv/lib/python3.13/site-packages (from ipython>=6.1.0->ipywidgets->leafmap->geoai-py) (4.9.0)\n",
      "Requirement already satisfied: prompt_toolkit<3.1.0,>=3.0.41 in ./.venv/lib/python3.13/site-packages (from ipython>=6.1.0->ipywidgets->leafmap->geoai-py) (3.0.50)\n",
      "Requirement already satisfied: pygments>=2.4.0 in ./.venv/lib/python3.13/site-packages (from ipython>=6.1.0->ipywidgets->leafmap->geoai-py) (2.19.1)\n",
      "Requirement already satisfied: stack_data in ./.venv/lib/python3.13/site-packages (from ipython>=6.1.0->ipywidgets->leafmap->geoai-py) (0.6.3)\n",
      "Requirement already satisfied: docstring-parser>=0.15 in ./.venv/lib/python3.13/site-packages (from jsonargparse[signatures]<5.0,>=4.27.7; extra == \"pytorch-extra\"->lightning[pytorch-extra]!=2.3.*,>=2->torchgeo->geoai-py) (0.16)\n",
      "Requirement already satisfied: typeshed-client>=2.1.0 in ./.venv/lib/python3.13/site-packages (from jsonargparse[signatures]<5.0,>=4.27.7; extra == \"pytorch-extra\"->lightning[pytorch-extra]!=2.3.*,>=2->torchgeo->geoai-py) (2.7.0)\n",
      "Requirement already satisfied: jsonschema-specifications>=2023.03.6 in ./.venv/lib/python3.13/site-packages (from jsonschema->flask-restx>=1.3.0->localtileserver->geoai-py) (2024.10.1)\n",
      "Requirement already satisfied: referencing>=0.28.4 in ./.venv/lib/python3.13/site-packages (from jsonschema->flask-restx>=1.3.0->localtileserver->geoai-py) (0.36.2)\n",
      "Requirement already satisfied: rpds-py>=0.7.1 in ./.venv/lib/python3.13/site-packages (from jsonschema->flask-restx>=1.3.0->localtileserver->geoai-py) (0.23.1)\n",
      "Requirement already satisfied: platformdirs>=2.5 in ./.venv/lib/python3.13/site-packages (from jupyter-core!=5.0.*,>=4.12->jupyter-server>=1.24.0->jupyter-server-proxy->geoai-py) (4.3.6)\n",
      "Requirement already satisfied: python-json-logger>=2.0.4 in ./.venv/lib/python3.13/site-packages (from jupyter-events>=0.11.0->jupyter-server>=1.24.0->jupyter-server-proxy->geoai-py) (3.3.0)\n",
      "Requirement already satisfied: rfc3339-validator in ./.venv/lib/python3.13/site-packages (from jupyter-events>=0.11.0->jupyter-server>=1.24.0->jupyter-server-proxy->geoai-py) (0.1.4)\n",
      "Requirement already satisfied: rfc3986-validator>=0.1.1 in ./.venv/lib/python3.13/site-packages (from jupyter-events>=0.11.0->jupyter-server>=1.24.0->jupyter-server-proxy->geoai-py) (0.1.1)\n",
      "Requirement already satisfied: bleach!=5.0.0 in ./.venv/lib/python3.13/site-packages (from bleach[css]!=5.0.0->nbconvert>=6.4.4->jupyter-server>=1.24.0->jupyter-server-proxy->geoai-py) (6.2.0)\n",
      "Requirement already satisfied: defusedxml in ./.venv/lib/python3.13/site-packages (from nbconvert>=6.4.4->jupyter-server>=1.24.0->jupyter-server-proxy->geoai-py) (0.7.1)\n",
      "Requirement already satisfied: jupyterlab-pygments in ./.venv/lib/python3.13/site-packages (from nbconvert>=6.4.4->jupyter-server>=1.24.0->jupyter-server-proxy->geoai-py) (0.3.0)\n",
      "Requirement already satisfied: mistune<4,>=2.0.3 in ./.venv/lib/python3.13/site-packages (from nbconvert>=6.4.4->jupyter-server>=1.24.0->jupyter-server-proxy->geoai-py) (3.1.2)\n",
      "Requirement already satisfied: nbclient>=0.5.0 in ./.venv/lib/python3.13/site-packages (from nbconvert>=6.4.4->jupyter-server>=1.24.0->jupyter-server-proxy->geoai-py) (0.10.2)\n",
      "Requirement already satisfied: pandocfilters>=1.4.1 in ./.venv/lib/python3.13/site-packages (from nbconvert>=6.4.4->jupyter-server>=1.24.0->jupyter-server-proxy->geoai-py) (1.5.1)\n",
      "Requirement already satisfied: fastjsonschema>=2.15 in ./.venv/lib/python3.13/site-packages (from nbformat>=5.3.0->jupyter-server>=1.24.0->jupyter-server-proxy->geoai-py) (2.21.1)\n",
      "Requirement already satisfied: munch in ./.venv/lib/python3.13/site-packages (from pretrainedmodels>=0.7.1->segmentation-models-pytorch>=0.2->torchgeo->geoai-py) (4.0.0)\n",
      "Requirement already satisfied: markdown-it-py>=2.2.0 in ./.venv/lib/python3.13/site-packages (from rich<14.0,>=12.3.0->lightning[pytorch-extra]!=2.3.*,>=2->torchgeo->geoai-py) (3.0.0)\n",
      "Requirement already satisfied: protobuf>=3.20 in ./.venv/lib/python3.13/site-packages (from tensorboardX<3.0,>=2.2->lightning[pytorch-extra]!=2.3.*,>=2->torchgeo->geoai-py) (6.30.1)\n",
      "Requirement already satisfied: ptyprocess in ./.venv/lib/python3.13/site-packages (from terminado>=0.8.3->jupyter-server>=1.24.0->jupyter-server-proxy->geoai-py) (0.7.0)\n",
      "Requirement already satisfied: soupsieve>1.2 in ./.venv/lib/python3.13/site-packages (from beautifulsoup4->gdown->leafmap->geoai-py) (2.6)\n",
      "Requirement already satisfied: httpcore==1.* in ./.venv/lib/python3.13/site-packages (from httpx->rio-tiler->localtileserver->geoai-py) (1.0.7)\n",
      "Requirement already satisfied: h11<0.15,>=0.13 in ./.venv/lib/python3.13/site-packages (from httpcore==1.*->httpx->rio-tiler->localtileserver->geoai-py) (0.14.0)\n",
      "Requirement already satisfied: PySocks!=1.5.7,>=1.5.6 in ./.venv/lib/python3.13/site-packages (from requests[socks]->gdown->leafmap->geoai-py) (1.7.1)\n",
      "Requirement already satisfied: webencodings in ./.venv/lib/python3.13/site-packages (from bleach!=5.0.0->bleach[css]!=5.0.0->nbconvert>=6.4.4->jupyter-server>=1.24.0->jupyter-server-proxy->geoai-py) (0.5.1)\n",
      "Requirement already satisfied: tinycss2<1.5,>=1.1.0 in ./.venv/lib/python3.13/site-packages (from bleach[css]!=5.0.0->nbconvert>=6.4.4->jupyter-server>=1.24.0->jupyter-server-proxy->geoai-py) (1.4.0)\n",
      "Requirement already satisfied: parso<0.9.0,>=0.8.4 in ./.venv/lib/python3.13/site-packages (from jedi>=0.16->ipython>=6.1.0->ipywidgets->leafmap->geoai-py) (0.8.4)\n",
      "Requirement already satisfied: fqdn in ./.venv/lib/python3.13/site-packages (from jsonschema[format-nongpl]>=4.18.0->jupyter-events>=0.11.0->jupyter-server>=1.24.0->jupyter-server-proxy->geoai-py) (1.5.1)\n",
      "Requirement already satisfied: isoduration in ./.venv/lib/python3.13/site-packages (from jsonschema[format-nongpl]>=4.18.0->jupyter-events>=0.11.0->jupyter-server>=1.24.0->jupyter-server-proxy->geoai-py) (20.11.0)\n",
      "Requirement already satisfied: jsonpointer>1.13 in ./.venv/lib/python3.13/site-packages (from jsonschema[format-nongpl]>=4.18.0->jupyter-events>=0.11.0->jupyter-server>=1.24.0->jupyter-server-proxy->geoai-py) (3.0.0)\n",
      "Requirement already satisfied: uri-template in ./.venv/lib/python3.13/site-packages (from jsonschema[format-nongpl]>=4.18.0->jupyter-events>=0.11.0->jupyter-server>=1.24.0->jupyter-server-proxy->geoai-py) (1.3.0)\n",
      "Requirement already satisfied: webcolors>=24.6.0 in ./.venv/lib/python3.13/site-packages (from jsonschema[format-nongpl]>=4.18.0->jupyter-events>=0.11.0->jupyter-server>=1.24.0->jupyter-server-proxy->geoai-py) (24.11.1)\n",
      "Requirement already satisfied: mdurl~=0.1 in ./.venv/lib/python3.13/site-packages (from markdown-it-py>=2.2.0->rich<14.0,>=12.3.0->lightning[pytorch-extra]!=2.3.*,>=2->torchgeo->geoai-py) (0.1.2)\n",
      "Requirement already satisfied: wcwidth in ./.venv/lib/python3.13/site-packages (from prompt_toolkit<3.1.0,>=3.0.41->ipython>=6.1.0->ipywidgets->leafmap->geoai-py) (0.2.13)\n",
      "Requirement already satisfied: cffi>=1.0.1 in ./.venv/lib/python3.13/site-packages (from argon2-cffi-bindings->argon2-cffi>=21.1->jupyter-server>=1.24.0->jupyter-server-proxy->geoai-py) (1.17.1)\n",
      "Requirement already satisfied: executing>=1.2.0 in ./.venv/lib/python3.13/site-packages (from stack_data->ipython>=6.1.0->ipywidgets->leafmap->geoai-py) (2.2.0)\n",
      "Requirement already satisfied: asttokens>=2.1.0 in ./.venv/lib/python3.13/site-packages (from stack_data->ipython>=6.1.0->ipywidgets->leafmap->geoai-py) (3.0.0)\n",
      "Requirement already satisfied: pure-eval in ./.venv/lib/python3.13/site-packages (from stack_data->ipython>=6.1.0->ipywidgets->leafmap->geoai-py) (0.2.3)\n",
      "Requirement already satisfied: pycparser in ./.venv/lib/python3.13/site-packages (from cffi>=1.0.1->argon2-cffi-bindings->argon2-cffi>=21.1->jupyter-server>=1.24.0->jupyter-server-proxy->geoai-py) (2.22)\n",
      "Requirement already satisfied: arrow>=0.15.0 in ./.venv/lib/python3.13/site-packages (from isoduration->jsonschema[format-nongpl]>=4.18.0->jupyter-events>=0.11.0->jupyter-server>=1.24.0->jupyter-server-proxy->geoai-py) (1.3.0)\n",
      "Requirement already satisfied: types-python-dateutil>=2.8.10 in ./.venv/lib/python3.13/site-packages (from arrow>=0.15.0->isoduration->jsonschema[format-nongpl]>=4.18.0->jupyter-events>=0.11.0->jupyter-server>=1.24.0->jupyter-server-proxy->geoai-py) (2.9.0.20241206)\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "%pip install geoai-py"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": ".venv",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.13.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
